initial commit

Lukease opened the pull request.

Commit:
initial commit

jpudysz commented on click/.gitignore:
```diff
@@ -22,3 +22,4 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+
```
.idea, style.css style.css.map to gitignore

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+app.appendChild(firstBox)
+const secondBox = document.createElement('div')
+secondBox.classList.add('app__box')
+app.appendChild(secondBox)
```
maybe create additional function like `createGameUI`?

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+const stopTime = document.createElement('p')
+stopTime.innerHTML = 'Stop Time'
+
+
```
remove extra enter

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+firstBox.appendChild(timer)
+
+const stopTime = document.createElement('p')
+stopTime.innerHTML = 'Stop Time'
```
this can be in the `createGameUI` function too

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+stopTime.innerHTML = 'Stop Time'
+
+
+let seconds = 2
```
why seconds = 2?

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+
+
+let seconds = 2
+let minutes = parseInt(seconds / 60)
```
valid syntax for `parseInt` is `parseInt(value, radix)`, radix should be 10

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+    if (seconds === 0 && minutes === 1) {
+        minutes -= 1
+        seconds = 60
+    } else if (seconds === 0) {
```
avoid `else`

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+let minutes = parseInt(seconds / 60)
+seconds = seconds % 60
+
+const interval = setInterval(function () {
```
please use `arrow function`

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+    }
+    seconds -= 1
+    return timer.innerHTML = `${minutes.toString().padStart(2, '0')}:` + seconds.toString().padStart(2, '0')
+}, 1000);
```
remove semicolon

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+        alert(`${points} : Total Score`)
+    }
+    seconds -= 1
+    return timer.innerHTML = `${minutes.toString().padStart(2, '0')}:` + seconds.toString().padStart(2, '0')
```
please don't add strings

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+secondBox.appendChild(secondBoxContainer)
+secondBoxContainer.classList.add('app__container')
+
+const circles = [1, 2, 3, 4, 5]
```
`Array.from(new Array(5))`

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+
+const allCircles = document.querySelectorAll('.circle')
+
+const addColor = setInterval(function () {
```
arrow function

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+    coloredCircle.classList.add('circle__red')
+
+    document.querySelector('.app__container').addEventListener('click', scorePoint)
+
```
remove enter

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,77 @@
+
+
+function scorePoint(event) {
+
```
remove enter, arrow function

Commit:
created new functions: 'createTimerBox', 'createCircleBox'.
removed extra enters,
change second to 60,
removed else,
add arrow function,
remove semicolon,
add idea, style.css , style.css.map to gitignore.

jpudysz commented on click/main.js:
```diff
@@ -1,72 +1,78 @@
+    const firstBox = document.createElement('div')
+    firstBox.classList.add('app__box')
+    app.appendChild(firstBox)
+    return document.querySelector('.app__box')
```
enter before return

jpudysz replied:
also, check comment below, you don't need query selector

jpudysz commented on click/main.js:
```diff
@@ -1,72 +1,78 @@
+    const secondBox = document.createElement('div')
+    secondBox.classList.add('app__box')
+    app.appendChild(secondBox)
+    return document.querySelectorAll('.app__box')[1]
```
aa (as above)

jpudysz replied:
why are you doing querySelector if you have reference to this element?

simply return:
```
return secondBox
```

jpudysz commented on click/main.js:
```diff
@@ -1,72 +1,78 @@
 const timer = document.createElement('div')
 timer.classList.add('timer')
-firstBox.appendChild(timer)
+createTimerBox().appendChild(timer)
```
you can move this code to the `createTimerBox` and return `timer`

jpudysz commented on click/main.js:
```diff
@@ -1,72 +1,78 @@
         seconds = 60
-    } else if (seconds === 0) {
+    }
+    if (seconds === 0) {
```
enter before `if`

jpudysz commented on click/main.js:
```diff
@@ -1,72 +1,78 @@
+    if (seconds === 0) {
         clearInterval(interval)
-        firstBox.appendChild(stopTime)
+        createTimerBox().appendChild(stopTime)
```
why are you creating it every second?

Commit:
add enter before return, if,
changes return in createTimerBox and createCircleBox,
add create timer to createTimerBox,
remove Stop Time,
add if to addColor

jpudysz commented on click/.gitignore:
```diff
@@ -22,3 +22,6 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+.idea/
```
you need to remove .idea` now from the project to make it work
This folder is hidden, so you need to show it somehow on windows

jpudysz commented on click/main.js:
```diff
@@ -0,0 +1,87 @@
+
+const scorePoint = event => {
+
+    if (event.target.classList.toString() === 'circle circle__red') {
```
`classList.includes()`

Commit:
change event.target.classList.toString() to event.target.classList.contains('circle__red')
